src/components/services/Services.tsx

```diff
@@ -123,6 +127,60 @@ export function Services(props: Props) {
     'Services 2'
   )
 
+  // Subscribe to Android Power Saver state, and show a warning only if it
+  // changes from off to on:
+  useAsyncEffect(
+    async () => {
+      let previousState = false
+      let isWarningShowing = false
+
+      const showBatterySaverWarning = async () => {
+        if (isWarningShowing) return
+
+        isWarningShowing = true
+        await Airship.show(bridge => <AlertDropdown bridge={bridge} message={lstrings.warning_battery_saver} warning persistent />)
+        isWarningShowing = false
+      }
+
+      const checkPowerSavingMode = async () => {
+        // This method is only available for Android
+        if (powerSavingOn != null) {
```

Comment:
Just check at the top of the useAsyncEffect

`if (Platform.OS !== 'android') return`

This eliminates all the checks for powerSavingOn or powerSavingModeChanged

Reply:
Those functions are defined as optional by the lib. I'd need to also patch the lib itself and this seemed like the simpler approach.